client/engine/engine.go

```diff
@@ -36,7 +36,11 @@ func (uce *ErrUnhandledChainEvent) Error() string {
 // Engine is the imperative part of the core business logic of a go-nitro Client
 type Engine struct {
 	// inbound go channels
-	FromAPI    chan APIEvent // This one is exported so that the Client can send API calls
+
+	// From API
```

Comment:
Outside the scope of this PR but are we using `chan` to communicate with the client API to allow us to easily scale to more engines?

Reply:
I think the pattern will allow us to handle concurrency easily. I don't know if that means more engines, necessarily -- perhaps more request handlers (running in goroutines). 

If we exposed the engine's handler functions directly and had the client call those synchronously, the client (and therefore the consuming application) would block until that handler had completed. Using a `chan` means the request can be dispatched asynchronously.